reindexer/reindex_worker/src/main/scala/uk/ac/wellcome/platform/reindex_worker/services/ReindexService.scala

```diff
@@ -56,35 +56,38 @@ class ReindexService @Inject()(dynamoDBClient: AmazonDynamoDB,
     // large, this might cause out-of-memory errors -- in practice, we're
     // hoping that the shards/individual records are small enough for this
     // not to be a problem.
-    val queryResults: List[Either[DynamoReadError, HybridRecord]] =
-      Scanamo.exec(dynamoDBClient)(
-        index.query(
-          'reindexShard -> reindexJob.shardId and
-            KeyIs('reindexVersion, LT, reindexJob.desiredVersion)
-        )
-      )
-
-    val outdatedRecords = queryResults.map {
-      case Left(err: DynamoReadError) => {
-        warn(s"Failed to read Dynamo records for $reindexJob: $err")
-        throw GracefulFailureException(
-          new RuntimeException(s"Error in the DynamoDB query: $err")
+    val futureResults: Future[List[Either[DynamoReadError, HybridRecord]]] =
+      Future {
+        Scanamo.exec(dynamoDBClient)(
+          index.query(
+            'reindexShard -> reindexJob.shardId and
+              KeyIs('reindexVersion, LT, reindexJob.desiredVersion)
+          )
         )
       }
-      case Right(r: HybridRecord) => r
+
+    val futureOutdatedRecords: Future[List[HybridRecord]] = futureResults.map {
+      results =>
+        results.map {
```

Comment:
It could be nicer to extract this inner block into a function to not have this map inside a map thing